ratelimit.go

```diff
@@ -118,14 +102,19 @@ func (t *limiter) Take() time.Time {
 
 	// If sleepFor is positive, then we should sleep now.
 	if t.sleepFor > 0 {
-		t.clock.Sleep(t.sleepFor)
+		t.timer.Stop()
+		t.timer.Reset(t.sleepFor)
```

Comment:
stopping and resetting a timer is racy, take a look at the documentation for how to stop and reset:
https://golang.org/pkg/time/#Timer.Reset
```
if !t.Stop() {
	<-t.C
}
t.Reset(d)
```

Reply:
Good catch, thanks!